include/lsst/afw/image/PhotoCalib.h

```diff
@@ -0,0 +1,426 @@
+     *
+     * @return     The flux in maggies and error (sigma).
+     */
+    Quantity countsToMaggies(double counts, double countsSigma,
+                             const afw::geom::Point<double, 2> &point,
+                             bool throwOnNegativeCounts=false) const;
+    Quantity countsToMaggies(double counts, double countsSigma,
+                             bool throwOnNegativeCounts=false) const;
+    //@}
+
+    /**
+     * Convert sourceRecord[countsField_counts] (ADU) at location
+     *             (sourceRecord.get('x'), sourceRecord.get('y')) (pixels) to maggies and maggie error.
+     *
+     * @param[in]  sourceRecord  The source record to get counts and position from.
+     * @param[in]  countsField   The counts field: Keys of the form "*_counts" and "*_countsSigma" must exist.
+     *                           For example: countsField = "PsfFlux" -> "PsfFlux_counts", "PsfFlux_countsSigma"
+     * @param[in]  throwOnNegativeCounts  Raise an exception when passed negative counts.
+     *
+     * @return     The flux in maggies and error (sigma) for this source.
+     */
+    Quantity countsToMaggies(const afw::table::SourceRecord &sourceRecord,
+                                              const std::string &countsField,
+                                              bool throwOnNegativeCounts=false) const;
+
+    //@{
+    /**
+     * Convert sourceCatalog[countsField_counts] (ADU) at locations
+     *             (sourceCatalog.get('x'), sourceCatalog.get('y')) (pixels) to maggies.
+     *
+     * @param[in]  sourceCatalog  The source catalog to get counts and position from.
+     * @param[in]  countsField    The counts field: Keys of the form "*_counts" and "*_countsSigma" must exist.
+     *                            For example: countsField = "PsfFlux" -> "PsfFlux_counts", "PsfFlux_countsSigma"
+     * @param[out] result         Pass a pre-allocated array to put the output in, instead of
+     *                            returning it.
+     * @param[in]  throwOnNegativeCounts  Raise an exception when passed negative counts.
+     *
+     * @return     The flux in maggies and error (sigma) for this source.
+     */
+    ndarray::Array<double,2,2> countsToMaggies(const afw::table::SourceCatalog &sourceCatalog,
+                                               const std::string &countsField,
+                                               bool throwOnNegativeCounts=false) const;
+    void countsToMaggies(const afw::table::SourceCatalog &sourceCatalog,
+                         const std::string &countsField,
+                         ndarray::Array<double,2,2> result,
+                         bool throwOnNegativeCounts=false) const;
+    //@}
+
+    /**
+     * Convert sourceCatalog[countsField_counts] (ADU) at locations
+     *             (sourceCatalog.get('x'), sourceCatalog.get('x')) (pixels) to maggies
+     *             and write the results back to sourceCatalog[outField_mag].
```

Comment:
If we're iterating over the sourceCatalog already, might as well set the values as we compute them.